src/metrics/metrics.go

```diff
@@ -272,6 +220,64 @@ func populateCustomMetrics(instanceEntity *integration.Entity, connection *conne
 	}
 }
 
+// metricsFromCustomQueryRow obtains a map of metrics from a row resulting from a custom query.
+// A particular metric can be configured either with:
+// - Specific columns in the query: metric_name, metric_type, metric_value
+// - The corresponding `query.Name` and `query.Type`
+// When both are defined, the query columns have precedence. Besides, if type is not defined it is automatically deteced.
+// The rest of the query columns are also taken as metrics/attributes (detecting their types automatically).
+// Besides, if `query.Prefix` is defined, all metric and attribute names will include the corresponding prefix.
+func metricsFromCustomQueryRow(row []string, columns []string, query customQuery) (map[string]customQueryMetricValue, error) {
+	metrics := map[string]customQueryMetricValue{}
+
+	var metricValue string
+	metricType := query.Type
+	metricName := query.Name
+
+	for i, columnName := range columns {
```

Comment:
I understood what you are doing but I took 2'. can you add a comment saying that you are scanning the columns one by one to place the right thing in the right place?

Reply:
I've just added additional comments, let me know if it is clearer now 🙂